state/state.go

```diff
@@ -27,3 +34,44 @@ type View interface {
 	NewView(ctx context.Context, changes merkledb.ViewChanges) (merkledb.View, error)
 	GetMerkleRoot(ctx context.Context) (ids.ID, error)
 }
+
+// StateKey holds the name of the key and its permission (Read/Write). By default,
+// initialization of Keys with duplicate key will not work. And to prevent duplicate
+// insertions from overriding the original permissions, use the Add function below.
+// Permission uses bits since transactions are expected to include this encoding
+type Keys map[string]Permission
+
+// The access permission (Read/Write) for each StateKey
+// Specifying RWrite is setting both the Read and Write bit
+type Permission set.Bits
+
+// By default, no permission bits are set
+func NewPermission(permissions ...uint8) Permission {
+	withinBoundsPermission := []int{} // NewBits requires int type
+	for _, v := range permissions {
+		// Only set bit to 1 if we're within 8 bits
```

Comment:
This seems like a fatal error that should not happen if the type system is respected. Nonetheless, I think the skip approach is correct.